packages/system/src/theme.js

```diff
@@ -0,0 +1,86 @@
+import { css } from 'styled-components';
+
+function resolve(obj, path) {
+  try {
+    return path
+      .split('.')
+      .reduce((prev, curr) => prev && prev[curr], obj)
+      .toString();
+  } catch {
+    return undefined;
+  }
+}
+
+const getFromTheme = props => spec => {
+  const {
+    theme: {
+      yoga: { [spec]: matchedSpec },
+    },
+  } = props;
+
+  return matchedSpec;
+};
+
+const getSpacing = props => getFromTheme(props)('spacing');
+const getBorder = props => getFromTheme(props)('borders');
+const getBorderRadius = props => getFromTheme(props)('radii');
+const getColor = props => getFromTheme(props)('colors');
```

Comment:
I may be missing something, but shouldn't this be `text.colors` (or something like that haha)? 🤔

Reply:
by getting from `colors.text` instead of `colors`, we won't be able to use all colors such as Vibin, Hope, Energy, and so on.

Getting from colors, we can get all of them plus all text colors, simply by doing:

```jsx
import { color } from '@gympass/yoga-system';
import styled from 'styled-components';

const Text = styled.p(color);

{/* works */}
<Text color="hope" /> 

{/* works too */}
<Text color="text.primary" /> 
```

So, with this in mind, I would rather get from `colors` to only `colors.text`.
What do you think?

Reply:
This decision has to be made by the design team, doesn't it? Should we be able to have texts on all these colors or should we limit what we can do?

If that's what we want then I don't mind. 😁

Reply:
We have the ok from the PD team. We're all done, so. Ok?